rest/index.py

```diff
@@ -468,6 +481,433 @@ def createIncident():
     return response
 
 
+@post("/alertstatus")
+@post("/alertstatus/")
+def update_alert_status():
+    """Update the status of an alert.
+
+    Requests are expected to take the following (JSON) form:
+
+    ```
+    {
+        "alert": str,
+        "status": str,
+        "user": {
+            "email": str,
+            "slack": str
+        },
+        "identityConfidence": str
+        "response": str
+    }
+    ```
+
+    Where:
+        * `"alert"` is the unique identifier fo the alert whose status
+        we are to update.
+        * `"status"` is one of "manual", "inProgress", "acknowledged"
+        or "escalated".
+        * `identityConfidence` is one of "highest", "high", "moderate", "low",
+        or "lowest".
+
+
+    This function writes back a response containing the following JSON.
+
+    ```
+    {
+        "error": Optional[str]
+    }
+    ```
+
+    If an error occurs and the alert is not able to be updated, then
+    the "error" field will contain a string message describing the issue.
+    Otherwise, this field will simply be `null`.  This function will,
+    along with updating the alert's status, append information about the
+    user and their response to `alert['details']['triage']`.
+
+    Responses will also use status codes to indicate success / failure / error.
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    alerts = mongo.meteor["alerts"]
+
+    try:
+        req = json.loads(request.body.read())
+        request.body.close()
+    except ValueError:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps({"error": "Missing or invalid request body"})
+        return response
+
+    valid_statuses = ["manual", "inProgress", "acknowledged", "escalated"]
+
+    if req["status"] not in valid_statuses:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps(
+            {"error": "Status not one of {}".format(" or ".join(valid_statuses))}
+        )
+        return response
+
+    valid_confidences = ["highest", "high", "moderate", "low", "lowest"]
+
+    if req["identityConfidence"] not in valid_confidences:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps(
+            {
+                "error": "user.identityConfidence not one of {}".format(
+                    " or ".join(valid_confidences)
+                )
+            }
+        )
+        return response
+
+    details = {
+        "triage": {"user": req["user"], "response": req["response"]},
+        "identityConfidence": req["identityConfidence"],
+    }
+
+    modified_count = 0
+
+    modified_count += alerts.update_one(
+        {"esmetadata.id": req["alert"]}, {"$set": {"status": req["status"]}}
```

Comment:
You may want to explicitly check for the presence of these fields to avoid a malformed JSON payload from causing an exception.